### PR TITLE
feat: use properly match with no custom distance func

### PR DESCRIPTION
**Pull request by JoanFM**

### CODE REVIEW

File: jinahub/indexers/SimpleIndexer/simple_indexer.py [hunk 1]
```diff
@@ -36,22 +35,21 @@ def __init__(
...
         self.default_top_k = default_top_k
-        if distance_metric == 'cosine':
-            self.distance = _cosine
-        elif distance_metric == 'euclidean':
-            self.distance = _euclidean
-        else:
-            raise ValueError('This distance metric is not available!')
+        self._distance = distance_metric
+        self._use_scipy = True
+        if distance_metric in ['cosine', 'euclidean', 'sqeuclidean']:
```
Comment (cristianmtr) [inline jinahub/indexers/SimpleIndexer/simple_indexer.py:40]: Can u throw an error if distance is not supported?

Comment (JoanFM) [reply jinahub/indexers/SimpleIndexer/simple_indexer.py:40]: we do not have an exhaustive list. It depends on what `scipy` supports.

Comment (JoanFM) [reply jinahub/indexers/SimpleIndexer/simple_indexer.py:40]: These ones are the ones supported natively, there are others offered via `scipy`

Comment (JoanFM) [reply jinahub/indexers/SimpleIndexer/simple_indexer.py:40]: I changed the docstrings

**Review by winstonww — APPROVED** [review]
LGTM

**Pull request merged**